Classes/Emogrifier.php

```diff
@@ -296,12 +260,59 @@ public function emogrify()
             $this->removeInvisibleNodes($xpath);
         }
 
-        $this->copyCssWithMediaToStyleNode($cssParts, $xmlDocument);
+        $this->copyCssWithMediaToStyleNode($cssParts['media'], $xmlDocument);
 
         return $xmlDocument->saveHTML();
     }
 
     /**
+     * Parse a list of selectors from a string of css.
+     *
+     * @param string[] $css a string of raw CSS code
+     *
+     * @return array
+     */
+    private function parseSelectors($css)
+    {
+        $cssKey = md5($css);
+        if (!isset($this->caches[self::CACHE_KEY_CSS][$cssKey])) {
+            // process the CSS file for selectors and definitions
+            preg_match_all('/(?:^|[\\s^{}]*)([^{]+){([^}]*)}/mis', $css, $matches, PREG_SET_ORDER);
```

Comment:
This comment is here because the preg match is not very clear. Please let's move it to a well-named private method instead.